free-1.md

```diff
@@ -1,68 +1,198 @@
+to. For example, if I draw a line from one angle of a triangle to the other
+side, am I drawing to the side's midpoint or bisecting the angle? These are
+usually not the same, but are close to each other!
+
+The tkz-euclid solution on the other hand, has a static textual interface. You
+can draw lines, draw shapes, and label intersections. The system is
+impressively complete, but seems cryptic at times. It is not clear how to do
+things like find an intersection, and more generally you don't always know what
+options are available to you. Furthermore, the names of the language correspond
+vaguely to the mathematical writing that they are based on, but are heavily
+constrained by LaTeX
+
+I would like to create an experience that allows for more complete interaction
+between the user and the computer. It would be awesome to allow for users to
+express what they would like to draw in a language and syntax that feels
+natural to them, while also getting visual feedback about what they're drawing
+and what they could choose to do next.
 
 ### Why a language?
-_Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+I think that I should address this question in two parts:
+   1. This question should be solved with a DSL because geometric writing is
+   already designed to unambiguously communicate geometric constructions. There
+   we should try to teach computers how to understand that language, or
+   something similar to it.
+   1. Why interactive? Is making something interactive really part of DSLs? Is
+   making something interactive really changing the language? I argue yes.
+   Language is fundamentally a framework for communication. To me it seems odd
+   that programming languages so far have just been about users telling
+   computers what to do. Allowing for computers to give feedback to users
+   expands the language to allow for 2-way communication. And the way the
+   computer sends back information will need to be designed well to make it
+   most useful to the users.
 
 ### Why you?
-_What excites you about this idea? How did you come up with it?_
 
+I've always enjoyed geometry, and am excited about building an interactive
+system which makes computers a better platform for exploring mathematical ideas
+that are not fully formed yet.
 
 ### Domain
-_Describe the project's domain in five words._
 
+Interactively creating geometric drawings
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
 
+Users could interact with the language statically or interactively.
+
+In either case they would write statements like:
+
+```
+circle x with center C
+line y tangent to C at point D
+line z through points C and D
+```
+
+I really like language idea because it mirrors the way that mathematicians
+write about geometry. Who hasn't seen something like:
```

Comment:
This is a good point!  It also isn't hard to imagine a parser that could read through your proof and, assuming you use some consistent and known style, generate related images for you when you need them.